### PR TITLE
XmlTest is null when read via IInvokedMethodListener

### LINKED ISSUE #1029: Issue with getting XmlTest from test method
HI,

Please found following [issue](http://stackoverflow.com/questions/36689568/testng-parallel-execution-behavior-doesnt-match-the-expected-one/36770699#36770699). There is an issue with getting XmlTest from InvokedMethod in case when test are running in parallel (for single run it works correctly). So there is `@Test(invocationCount = 5, threadPoolSize = 5)` and `invokedMethod.getTestMethod().getXmlTest()` return null but `testResult.getTestContext().getCurrentXmlTest()` returns XmlTest correctly. It seems like a bug...

### PR DESCRIPTION
**Pull request by krmahadevan**
Closes #1029

Fixes #1029  .

### Did you remember to?

- [X] Add test case(s)
- [X] Update `CHANGES.txt`

We encourage pull requests that:

* Add new features to TestNG (or)
* Fix bugs in TestNG

If your pull request involves fixing SonarQube issues then we would suggest that you please discuss this with the 
[TestNG-dev](https://groups.google.com/forum/#!forum/testng-dev) before you spend time working on it.

### CODE REVIEW

**Comment by krmahadevan** [comment]
ping @juherr - Please help take a look

File: src/test/java/test/listeners/ListenerTest.java [hunk 1]
```diff
@@ -260,4 +267,31 @@ public void testFailATestFromOnTestStart() {
...
+  @Test(dataProvider = "dp", description ="GITHUB-1029" )
+  public void ensureXmlTestIsNotNull(Class<?> clazz, XmlSuite.ParallelMode mode) {
+    XmlSuite xmlSuite = createXmlSuite("Suite");
+    createXmlTest(xmlSuite, "GITHUB-1029-Test", clazz);
+    xmlSuite.setParallel(mode);
+    Issue1029InvokedMethodListener listener = new Issue1029InvokedMethodListener();
+    TestNG testng = create(xmlSuite);
+    testng.addListener((ITestNGListener) listener);
+    testng.setThreadCount(10);
+    testng.setThreadCount(10);
```
Comment (juherr) [inline src/test/java/test/listeners/ListenerTest.java]: duplicate line

**Review by juherr — APPROVED** [review]
LGTM 👍

File: src/test/java/test/listeners/ListenerTest.java (hunk 1 above)
Comment (krmahadevan) [reply src/test/java/test/listeners/ListenerTest.java]: Oops.. Copy paste error. Fixed and pushed.

Comment (juherr) [reply src/test/java/test/listeners/ListenerTest.java]: 👍

**Review by cbeust — COMMENTED** [review]
LGTM, feel free to merge after resolving the conflicts.

**Pull request merged**